types/src/applet.ts

```diff
@@ -20,9 +21,8 @@ export interface AppletConfig {
     }
 }
 
-export interface AppletConfigInput {
+export type AppletConfigInput = CommonConfig & {
     name: string,
-    // pub ranges: Array<Range>, // leaving out ranges since this is not an entry and is just part of the dimension
```

Comment:
This was one of the feedbacks we received from Damien, to have Ranges as entries. I think @tatssato is working on coding this in, no? @tatssato do you remember why we needed Ranges as entries?

Reply:
It was so they could be fetched as well as reused.

Reply:
Yup exactly what @weswalla said. I will be working on this in a separate PR including other recommendations Damien had :D

Reply:
@tatssato note that there is a PR for beta-rc.2 with passing local tests but something about nix-shell in github CI isn't working currently, you may want to work off a branch of that, or merge it into your current branch!

Reply:
@weswalla Thanks. Got it. Will merge the beta-rc.2 to this branch :D